### PR TITLE
add tests for blitframebuffer when width or height of src/dst region overflow

### PR DESCRIPTION
**Pull request by Richard-Yunchao**
Add some test cases per the suggestion at https://codereview.chromium.org/2409523002/#msg43 by Zhenyao. 

PTAL. Thanks a lot! @zhenyao , @kenrussell .

### CODE REVIEW

**Comment by zhenyao** [comment]
Actually the working group consensus is not to enforce a GL error in such overflow cases.  Just call blitFramebuffer() and if the test is able to finish without crash, then it's OK.  I suggest to test that in a separate html (so if it crashes, it won't affect other test cases in the same html)

**Comment by Richard-Yunchao** [comment]
That's true, generate INVALID_OPERATION is not a must. On many devices, the underlying drivers can handle this situation without any error. It can be NO_ERROR too. 
@zhenyao , move the test to a standalone file as you suggested too. Please take another look.

File: sdk/tests/conformance2/rendering/00_test_list.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 attrib-type-match.html
 blitframebuffer-filter-outofbounds.html
 blitframebuffer-filter-srgb.html
+blitframebuffer-misc.html
```
Comment (zhenyao) [inline sdk/tests/conformance2/rendering/00_test_list.txt]: maybe blitframebuffer-size-overflow.html

File: sdk/tests/conformance2/rendering/blitframebuffer-misc.html [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+</head>
+<body>
+<canvas id="example" width="8" height="8"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+
+var wtu = WebGLTestUtils;
+description("This test verifies the functionality of blitFramebuffer for some corner cases.");
```
Comment (zhenyao) [inline sdk/tests/conformance2/rendering/blitframebuffer-misc.html]: Maybe "This test verifies blitFramebuffer won't cause a crash when the computed sizes might overflow."

File: sdk/tests/conformance2/rendering/blitframebuffer-misc.html [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    gl.framebufferTexture2D(gl.READ_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, tex1, 0);
+    if ((gl.checkFramebufferStatus(gl.READ_FRAMEBUFFER) != gl.FRAMEBUFFER_COMPLETE) ||
+        (gl.checkFramebufferStatus(gl.DRAW_FRAMEBUFFER) != gl.FRAMEBUFFER_COMPLETE)) {
+        testFailed("Framebuffer incomplete.");
+        return;
+    }
+
+    var max = 0x7fffffff;
+    gl.blitFramebuffer(0, 0, max, max, 0, 0, width, height, gl.COLOR_BUFFER_BIT, gl.NEAREST);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "BlitFramebuffer should succeed.");
```
Comment (zhenyao) [inline sdk/tests/conformance2/rendering/blitframebuffer-misc.html]: I feel these are extreme cases and there is no need for us to make sure it works.  Let's just not check GL errors.

Comment (zhenyao) [inline sdk/tests/conformance2/rendering/blitframebuffer-misc.html]: Same below.

File: sdk/tests/conformance2/rendering/blitframebuffer-misc.html [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    gl.blitFramebuffer(0, 0, width, height, 0, 0, max, max, gl.COLOR_BUFFER_BIT, gl.NEAREST);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "BlitFramebuffer should succeed.");
+    gl.blitFramebuffer(0, 0, max, max, 0, 0, max, max, gl.COLOR_BUFFER_BIT, gl.NEAREST);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "BlitFramebuffer should succeed.");
+
+    gl.blitFramebuffer(-1, -1, max, max, 0, 0, width, height, gl.COLOR_BUFFER_BIT, gl.NEAREST);
+    wtu.glErrorShouldBe(gl, [gl.NO_ERROR, gl.INVALID_OPERATION], "The width and height of src region may overflow.");
```
Comment (zhenyao) [inline sdk/tests/conformance2/rendering/blitframebuffer-misc.html]: Same here, I think if an implementation wants to generate INVALID_VALUE or OUT_OF_MEMORY, that's also fine.  Let's just not check GL errors here.

**Comment by kenrussell** [comment]
Agree with Mo's observations; I defer review to him.

File: sdk/tests/conformance2/rendering/blitframebuffer-misc.html (hunk 4 above)
Comment (Richard-Yunchao) [inline sdk/tests/conformance2/rendering/blitframebuffer-misc.html]: Thanks for your review, Zhenyao. So, do you have some suggestions to do the check? I mean, what behavior is correct, what is false? @zhenyao.

File: sdk/tests/conformance2/rendering/00_test_list.txt (hunk 1 above)
Comment (Richard-Yunchao) [inline sdk/tests/conformance2/rendering/00_test_list.txt]: Done

File: sdk/tests/conformance2/rendering/blitframebuffer-misc.html (hunk 4 above)
Comment (Richard-Yunchao) [inline sdk/tests/conformance2/rendering/blitframebuffer-misc.html]: Make sense. I updated the code accordingly. @zhenyao. could you take another look?

**Comment by zhenyao** [comment]
Thanks. LGTM.

**Comment by Richard-Yunchao** [comment]
Thanks for merging. Zhenyao.
